tests/test_fast.py

```diff
@@ -0,0 +1,91 @@
+from nose.tools import (
+    eq_,
+    set_trace,
+)
+from contextlib import contextmanager
+from io import BytesIO
+import os
+
+from fast import (
+    FASTNames,
+    LCSHNames,
+)
+
+BASE_DIR = os.path.split(__file__)[0]
+
+class MockFASTNames(FASTNames):
+    """Works just like FASTNames, but writes consolidated output to a
+    BytesIO object rather than a file on disk.
+    """
+
+    def __init__(self):
+        self.output_consolidated_file = None
+
+    @contextmanager
+    def consolidated_output_filehandle(self, path):
+        # Create a BytesIO object to stand in for the
+        # consolidated file that would otherwise be written to
+        # disk.
+        self.output_consolidated_file = BytesIO()
+        yield self.output_consolidated_file
+
+class MockLCSHNames(MockFASTNames, LCSHNames):
+    """Works just like LCSHNames, but writes consolidated output to a
+    BytesIO object rather than a file on disk.
+    """
+
+class TestFASTNames(object):
+
+    def test_from_data_directory_not_consolidated(self):
+        # Load FAST data from a number of gzipped files in N-Triples
+        # format.
+        not_consolidated = MockFASTNames.from_data_directory(
+            os.path.join(BASE_DIR, "files/fast/not-consolidated")
+        )
+        eq_(
+            {
+                '1726280': 'Filmed roundtables',
+                '631903': 'New Yorker (Fireboat)',
+                '1750175': 'Short stories, American'
+            },
+            not_consolidated
+        )
+        
+        # A consolidated file was written to "disk" in CSV format.
+        output = not_consolidated.output_consolidated_file
+        expect = '1726280,Filmed roundtables\r\n631903,New Yorker (Fireboat)\r\n1750175,"Short stories, American"\r\n'
+        eq_(expect, output.getvalue())
+
+    def test_from_data_directory_consolidated(self):
+        # Load FAST data from a single CSV file created by an earlier call
+        # to write_consolidated_file().
+        consolidated = MockFASTNames.from_data_directory(
+            os.path.join(BASE_DIR, "files/fast/consolidated")
+        )
+        eq_({'identifier1': 'name1', 'identifier2': 'name2'},
+            consolidated)
+
+        # Since data was loaded in from a consolidated file,
+        # no new consolidated file was created.
+        eq_(None, consolidated.output_consolidated_file)
+
+class TestLCSHNames(object):
+
+    def test_from_data_directory_not_consolidated(self):
+        # Load FAST data from a number of ZIP archives containing data
+        # in N-Triples format.
+        not_consolidated = MockLCSHNames.from_data_directory(
+            os.path.join(BASE_DIR, "files/fast/not-consolidated")
+        )
+        eq_(
+            {
+                'sj00001253': 'Ceratopsians',
+                'gf2008025611': 'Abstract films'
```

Comment:
Is one of the differences between FAST and LSCH that LSCH has letters in the identifiers?

Reply:
FAST doesn't have letters in the identifiers, but I don't know what the letters mean in LCSH identifiers. They're just different systems, although FAST is based on LCSH.

Looking it up, I see that "sj" means a juvenile subject heading. So it's the faceting information. In some cases we could derive some information from the FAST identifiers that we couldn't get from the names. "sj" means this is a book for kids (though we have no idea how old the kids are).

Reply:
The regex for each system is also different so it seems they get the headings from different sources.

Reply:
Yeah, they come from different orgs -- I think LSCH comes from Library of Congress and FAST comes from OCLC.